src/Middleware/CasProxyTicketMiddleware.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace OpenEuropa\EPoetry\Middleware;
+
+use Http\Promise\Promise;
+use Phpro\SoapClient\Middleware\Middleware;
+use Phpro\SoapClient\Middleware\MiddlewareInterface;
+use Psr\Http\Message\RequestInterface;
+
+class CasProxyTicketMiddleware extends Middleware implements MiddlewareInterface
+{
+    /**
+     * The Proxy Ticket to add to requests.
+     *
+     * @var string
+     */
+    protected $proxyTicket;
+
+    /**
+     * CasProxyTicketMiddleware constructor.
+     *
+     * @param string $proxyTicket
+     */
+    public function __construct(string $proxyTicket)
+    {
+        $this->proxyTicket = $proxyTicket;
```

Comment:
I'm not sure we can inject the ticket string in the service definition, shouldn't we inject the session handler and then get the ticket from that instead?

Reply:
Changed.